src/Bonsai.ML.Torch/ToNDArray.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Xml.Serialization;
+using System.Linq.Expressions;
+using System.Reflection;
+using Bonsai.Expressions;
+using static TorchSharp.torch;
+
+namespace Bonsai.ML.Torch
+{
+    /// <summary>
+    /// Converts the input tensor into an array of the specified element type and rank.
+    /// </summary>
+    [Combinator]
+    [Description("Converts the input tensor into an array of the specified element type.")]
+    [WorkflowElementCategory(ElementCategory.Transform)]
+    public class ToNDArray : SingleArgumentExpressionBuilder
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ToNDArray"/> class.
+        /// </summary>
+        public ToNDArray()
+        {
+            Type = typeof(double);
+        }
+
+        /// <summary>
+        /// Gets or sets the type mapping used to convert the input tensor into an array.
+        /// </summary>
+        [Description("Gets or sets the type mapping used to convert the input tensor into an array.")]
+        [TypeConverter(typeof(ScalarTypeConverter))]
+        [XmlIgnore]
+        public Type Type { get; set; }
```

Comment:
Same question as above, could we use directly `ScalarType` here, and then convert to a concrete `Type` during the `Build` step?